get_github_coderabbit_review/test_github_api_refactor.py

```
@@ -0,0 +1,302 @@
+
+        try:
+            client = GitHubClient()
+        except Exception as e:
+            print(f"❌ GitHubClient initialization failed: {e}")
+            return False
+
+    # Test invalid URL
+    try:
+        client.parse_pr_url("invalid-url")
+        print("❌ Should have raised InvalidPRUrlError")
+        return False
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)